examples/game-of-life/src/lib.rs

```diff
@@ -178,20 +177,21 @@ pub fn run() {
 
     // Kick off a loop that keeps computing a tick of the universe and then
     // re-rendering on every animation frame.
-    let rc: Rc<RefCell<Option<Closure<FnMut()>>>> = Rc::new(RefCell::new(None));
+    let rc = <Rc<RefCell<Option<Closure<dyn FnMut()>>>>>::default();
     let rc2 = rc.clone();
     let window2 = window.clone();
     let weak = vdom.weak();
     let f = Closure::wrap(Box::new(move || {
         weak.schedule_render();
 
-        wasm_bindgen_futures::spawn_local(
-            weak.with_component(|root| {
-                let universe = root.unwrap_mut::<Universe>();
-                universe.tick();
-            })
-            .map_err(|_| wasm_bindgen::throw_str("impossible, we always keep the vdom alive")),
-        );
+        // // -- tricky bit --
+        // wasm_bindgen_futures::spawn_local(
+        //     weak.with_component(|root| {
+        //         let universe = root.unwrap_mut::<Universe>();
+        //         universe.tick();
+        //     })
+        //     .map_err(|_| wasm_bindgen::throw_str("impossible, we always keep the vdom alive")),
+        // );
```

Comment:
`wasm_bindgen::spawn_local` requires a 'static lifetime (supposedly, this has changed in recent versions?), I found no easy way to change how this works

Reply:
I am pretty sure that `spawn_local` has always required `'static` futures. This is because ownership of the future is given to the JS engine to drive it to completion. Therefore the existing future spawned here should already by `'static` &mdash; what issues are arising here?

Reply:
You're right, it always did, my bad.
Replaced with:
```rust
        wasm_bindgen_futures::spawn_local(
            weak.with_component(|root| {
                let universe = root.unwrap_mut::<Universe>();
                universe.tick();
            })
            .map(|f| {
                if let Err(_) = f {
                    wasm_bindgen::throw_str("impossible, we always keep the vdom alive");
                }
            }),
        );
```
Causes the error:
```
error[E0495]: cannot infer an appropriate lifetime for autoref due to conflicting requirements
   --> examples/game-of-life/src/lib.rs:190:18
    |
190 |             weak.with_component(|root| {
    |                  ^^^^^^^^^^^^^^
    |
note: first, the lifetime cannot outlive the lifetime '_ as defined on the body at 185:36...
   --> examples/game-of-life/src/lib.rs:185:36
    |
185 |     let f = Closure::wrap(Box::new(move || {
    |                                    ^^^^^^^
note: ...so that closure can access `weak`
   --> examples/game-of-life/src/lib.rs:190:13
    |
190 |             weak.with_component(|root| {
    |             ^^^^
    = note: but, the lifetime must be valid for the static lifetime...
note: ...so that the type `futures_util::future::future::map::Map<impl core::future::future::Future, [closure@examples/game-of-life/src/lib.rs:194:18: 198:14]>` will meet its required lifetime bounds
   --> examples/game-of-life/src/lib.rs:189:9
    |
189 |         wasm_bindgen_futures::spawn_local(
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
Replaced with:
```rust
        wasm_bindgen_futures::spawn_local(async move {
            let fut = weak.with_component(|root| {
                let universe = root.unwrap_mut::<Universe>();
                universe.tick();
            });

            if let Err(_) = fut.await {
                wasm_bindgen::throw_str("impossible, we always keep the vdom alive");
            }
        });
```
Causes the error:
```
error[E0277]: expected a `std::ops::FnMut<()>` closure, found `[closure@examples/game-of-life/src/lib.rs:185:36: 209:6 weak:dodrio::vdom::VdomWeak, window:web_sys::Window, rc:std::rc::Rc<std::cell::RefCell<std::option::Option<wasm_bindgen::closure::Closure<dyn std::ops::FnMut()>>>>]`
   --> examples/game-of-life/src/lib.rs:185:27
    |
185 |       let f = Closure::wrap(Box::new(move || {
    |  ___________________________^
186 | |         weak.schedule_render();
187 | |
188 | |         // -- tricky bit --
...   |
208 | |             .unwrap_throw();
209 | |     }) as Box<dyn FnMut()>);
    | |______^ expected an `FnMut<()>` closure, found `[closure@examples/game-of-life/src/lib.rs:185:36: 209:6 weak:dodrio::vdom::VdomWeak, window:web_sys::Window, rc:std::rc::Rc<std::cell::RefCell<std::option::Option<wasm_bindgen::closure::Closure<dyn std::ops::FnMut()>>>>]`
    |
    = help: the trait `std::ops::FnMut<()>` is not implemented for `[closure@examples/game-of-life/src/lib.rs:185:36: 209:6 weak:dodrio::vdom::VdomWeak, window:web_sys::Window, rc:std::rc::Rc<std::cell::RefCell<std::option::Option<wasm_bindgen::closure::Closure<dyn std::ops::FnMut()>>>>]`
    = note: wrap the `[closure@examples/game-of-life/src/lib.rs:185:36: 209:6 weak:dodrio::vdom::VdomWeak, window:web_sys::Window, rc:std::rc::Rc<std::cell::RefCell<std::option::Option<wasm_bindgen::closure::Closure<dyn std::ops::FnMut()>>>>]` in a closure with no arguments: `|| { /* code */ }
    = note: required for the cast to the object type `dyn std::ops::FnMut()`
```

Reply:
I must admit that I'm going at this by trial and error and I haven't fully understood all the fundamentals of lifetimes and async yet (which is the reason why I attempted this in the first place), I hope I'm not wasting your time.

Reply:
Oh, I got it lol, I should have paid more attention reading the `todomvc` code, all I needed was to clone the `Rc`.